src/core/execute.c

```diff
@@ -3148,6 +3148,16 @@ static int exec_child(
                         return log_unit_error_errno(unit, errno, "Failed to set up CPU affinity: %m");
                 }
 
+        if (mpol_is_valid(context->numa_policy.type)) {
+                r = apply_numa_policy(&context->numa_policy);
+                if (r == -EOPNOTSUPP)
+                        log_unit_debug_errno(unit, SYNTHETIC_ERRNO(EOPNOTSUPP), "NUMA support not available, ignoring.");
```

Comment:
the error is not synthethic here, you just received it. Just replace `SYNTHETIC_ERRNO(EOPNOTSUPP)` with a simple `r`